### DISCUSSION TITLE
How to migrate the content of presets?

### DISCUSSION THREAD

**Discussion by ccopsey** [renovatebot/renovate#21779] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.58.0

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

My setup, which I have developed over time and am happy with, is to place minimal `renovate.json` files in my repos which extend a dedicated preset in a `Renovate` repo in the same project.  These dedicated presets in turn extend local/project specific presets (e.g. authentication hostrules) or common presets in a seperate project.  As I chop and change naming conventions for presets I end up with deprecated presets simply extending new presets.  Lots and lots of extending, but it works _really_ well.

Over the last couple of weeks there have been a number of config migrations in renovate that I know I need to implement.  `azureAutoApprove` for example; I can see it no longer appears in my "full resolved config" despite being in my presets.

I don't have any log messages telling me that my config needs migrating (even in debug).

Is there a process by which I can be notified that the content of my presets need attention?

### What you tried so far.

The logs make no mention of migrating my presets.  Other than scouring the renovate release notes, I'm not sure how else I could know what action to take.

**Comment by viceice** [comment]
automatically migrate presets is currently not supported

**Reply by rarkins** [reply]
Sounds like a good feature to have though?

Maybe:
- Renovate needs to be installed into the preset's repo
- New Renovate field to indicate which files are preset files, so that Renovate knows to check and migrate them?

**Reply by viceice** [reply]
sounds good 🤗 

can be glob or regex

**Reply by rarkins** [reply]
Trying to think of a good name for this which doesn't confuse people. e.g. it can't be "includePresets". Maybe "validatePresets"? Should it validate and migrate?

**Reply by ccopsey** [reply]
Hello both, thanks for the positive engagement.  From my perspective I would be perfectly happy with warnings similar to the existing "config needs migration" warnings.  So validation only.  Migration would be a significant bonus.

**Reply by viceice** [reply]
validation is already position with running the config validator manually

**Reply by ccopsey** [reply]
I was unaware of that tool, and that is a great start.  It would be nice if we could leverage the iterating over projects/repos and the clone/checkout processes that already occur to get me access to my json files.

**Reply by viceice** [reply]
you can checkout our .github repo workflow. there we've validation

**Reply by ccopsey** [reply]
I have been trying to use the validator app. I rapidly came up against the problem that it doesn't work with private presets however I have subsequently implemented that (works for me - might require a tidy). However, it appears the validator doesn't consider presets at the root level.  They are downloaded and analyzed when they are specified in packagerules, but not in a first level "extends".  Is this expected?

**Reply by viceice** [reply]
yes

**Reply by childnode** [reply]
> New Renovate field to indicate which files are preset files, so that Renovate knows to check and migrate them?

as far as I can see, the migration feature https://github.com/renovatebot/renovate/issues/1502 << https://github.com/renovatebot/renovate/pull/15122 is not implemented as manager

but would be helpful to maintain a preset repository on self-instance, wouldn't it?
